### PR TITLE
FRW-9135 Added AI tools how-to.

### PR DESCRIPTION
**Pull request by kraal-spryker**
Ticket: https://spryker.atlassian.net/browse/FRW-9135

### CODE REVIEW

File: docs/dg/dev/ai-assistants-getting-started-guide.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+- Cursor
+
+### Install GitHub Copilot
+
+GitHub Copilot is an AI pair programmer that helps you write code faster. It draws context from comments and code, and suggests individual lines and whole functions instantly.
+To start using GitHub Copilot, you need to perform few steps:
+- Install the GitHub Copilot [extension](https://plugins.jetbrains.com/plugin/17718-github-copilot) in your IDE.
+- Authorize yourself as a GitHub user in the Preferences->Languages and Frameworks-> GitHub Copilot.
+
+
```
Comment (kraal-spryker) [inline docs/dg/dev/ai-assistants-getting-started-guide.md:27]: CleanShot 2024-10-15 at 16 35 19@2x

File: docs/dg/dev/ai-assistants-getting-started-guide.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+And after these small steps, you will be able to use the GitHub Copilot.
+
+### Install Qodo
+
+The setup of the Qodo tool is straightforward. All you need to do is install the following [plugin](https://plugins.jetbrains.com/plugin/21206-qodo-gen-formerly-codiumate-). If the installation was correct, you could check that the plugin works. An example is in the screenshot below.
+
+
```
Comment (kraal-spryker) [inline docs/dg/dev/ai-assistants-getting-started-guide.md:36]: CleanShot 2024-10-15 at 16 36 13@2x

File: docs/dg/dev/ai-assistants-getting-started-guide.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+### Install Cursor
+
+Cursor is an AI powered IDE which is basically a fork from VS Code. To install Cursor, you need to follow the steps above:
+- You need to download the installation file from the [official website](https://www.cursor.com/). Once downloaded, run the installer to begin.
+- After the successful installation, you need to enable autosave files in the settings to prevent data loss during work. This will ensure your progress is saved automatically.
+
+
```
Comment (kraal-spryker) [inline docs/dg/dev/ai-assistants-getting-started-guide.md:44]: CleanShot 2024-10-15 at 16 36 48@2x

File: docs/dg/dev/ai-assistants-getting-started-guide.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+- Simplify the selected code fragment.
+- Generate tests for the selected code fragment.
+- IO interaction.
+
+### Code completion tool
+
+GitHub Copilot can generate suggestions based on the existing code inside your project.
+The video screenshot will show how it works:
+
+
```
Comment (kraal-spryker) [inline docs/dg/dev/ai-assistants-getting-started-guide.md:65]: CleanShot 2024-10-16 at 12 32 04@2x

File: docs/dg/dev/ai-assistants-getting-started-guide.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+
+### Describe the selected code fragment
+
+GitHub Copilot can explain the code fragment. The screenshot below will show this in action:
+
+
```
Comment (kraal-spryker) [inline docs/dg/dev/ai-assistants-getting-started-guide.md:72]: CleanShot 2024-10-16 at 12 33 56@2x

File: docs/dg/dev/ai-assistants-getting-started-guide.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
```
Comment (kraal-spryker) [inline docs/dg/dev/ai-assistants-getting-started-guide.md:73]: CleanShot 2024-10-16 at 12 35 06@2x

File: docs/dg/dev/ai-assistants-getting-started-guide.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+### Generate doc blocks for the selected code fragment
+
+GitHub Copilot can generate doc blocks for a method, as well as add comments to the code. The screenshot below will show this in action.
+
+
```
Comment (kraal-spryker) [inline docs/dg/dev/ai-assistants-getting-started-guide.md:79]: CleanShot 2024-10-16 at 12 36 29@2x

File: docs/dg/dev/ai-assistants-getting-started-guide.md [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
```
Comment (kraal-spryker) [inline docs/dg/dev/ai-assistants-getting-started-guide.md:80]: CleanShot 2024-10-16 at 12 37 53@2x

File: docs/dg/dev/ai-assistants-getting-started-guide.md [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+
+### Fix syntax bugs for the selected code fragment
+
+GitHub Copilot can suggest how to fix syntax bugs in the method. The screenshot below will show an example:
+
+
```
Comment (kraal-spryker) [inline docs/dg/dev/ai-assistants-getting-started-guide.md:86]: CleanShot 2024-10-16 at 12 38 55@2x

File: docs/dg/dev/ai-assistants-getting-started-guide.md [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+
```
Comment (kraal-spryker) [inline docs/dg/dev/ai-assistants-getting-started-guide.md:87]: CleanShot 2024-10-16 at 12 40 56@2x

File: docs/dg/dev/ai-assistants-getting-started-guide.md [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+
+### Simplify the selected code fragment
+
+GitHub Copilot can refactor a code fragment. The screenshot below will show this in action:
+
+
```
Comment (kraal-spryker) [inline docs/dg/dev/ai-assistants-getting-started-guide.md:93]: CleanShot 2024-10-16 at 12 56 32@2x

File: docs/dg/dev/ai-assistants-getting-started-guide.md [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+
```
Comment (kraal-spryker) [inline docs/dg/dev/ai-assistants-getting-started-guide.md:94]: CleanShot 2024-10-16 at 12 57 35@2x

File: docs/dg/dev/ai-assistants-getting-started-guide.md [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+
+### Generate tests for the selected code fragment
+
+GitHub Copilot can generate tests for the selected code fragment. Video below will show the example:
+
+
```
Comment (kraal-spryker) [inline docs/dg/dev/ai-assistants-getting-started-guide.md:100]: CleanShot 2024-10-16 at 12 58 27@2x

File: docs/dg/dev/ai-assistants-getting-started-guide.md [hunk 14]
```diff
@@ -0,0 +1,128 @@
...
+
```
Comment (kraal-spryker) [inline docs/dg/dev/ai-assistants-getting-started-guide.md:101]: CleanShot 2024-10-16 at 12 59 24@2x

File: docs/dg/dev/ai-assistants-getting-started-guide.md [hunk 15]
```diff
@@ -0,0 +1,128 @@
...
+### IO interaction
+
+You can interact with GitHub Copilot using prompts. An example is in the screenshot below.
+
+In the next example, you'll see how to generate transfer definition XML code using GitHub Copilot IO interaction mode.
+
+
```
Comment (kraal-spryker) [inline docs/dg/dev/ai-assistants-getting-started-guide.md]: CleanShot 2024-10-16 at 13 06 58@2x

File: docs/dg/dev/ai-assistants-getting-started-guide.md [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+GitHub Copilot can generate tests for the selected code fragment. The screenshot below will show the example:
+
+
+
+
+### IO interaction using GitHub Copilot
+
+You can interact with GitHub Copilot using prompts. An example is in the screenshot below.
+
+
```
Comment (kraal-spryker) [inline docs/dg/dev/ai-assistants-getting-started-guide.md:108]: CleanShot 2024-10-16 at 13 11 28@2x

File: docs/dg/dev/ai-assistants-getting-started-guide.md [hunk 17]
```diff
@@ -0,0 +1,128 @@
...
+
+### Generate tests using Qodo
+
+Alternatively, tests can be generated using the Qodo tool (formerly Codium). Qodo offers more flexible configuration options and a wider range of test scenarios compared to GitHub Copilot.
+The example is in the screenshot below:
+
+
```
Comment (kraal-spryker) [inline docs/dg/dev/ai-assistants-getting-started-guide.md:115]: CleanShot 2024-10-16 at 13 15 08@2x

File: docs/dg/dev/ai-assistants-getting-started-guide.md [hunk 18]
```diff
@@ -0,0 +1,128 @@
...
+
```
Comment (kraal-spryker) [inline docs/dg/dev/ai-assistants-getting-started-guide.md:116]: CleanShot 2024-10-16 at 13 16 28@2x

File: docs/dg/dev/ai-assistants-getting-started-guide.md [hunk 19]
```diff
@@ -0,0 +1,128 @@
...
+### Code generation using Cursor
+
+Cursor AI excels at offering context-aware suggestions by considering the bigger picture within the codebase. This enables it to deliver more accurate and well-structured recommendations, in contrast to GitHub Copilot, which mainly focuses on real-time suggestions based on the specific line of code you're working on.
+It's great at generating basic code and significantly saves time, especially when you need to create something from scratch.
+An example is in the screenshot below:
+
+
```
Comment (kraal-spryker) [inline docs/dg/dev/ai-assistants-getting-started-guide.md:123]: CleanShot 2024-10-16 at 16 44 39@2x

File: docs/dg/dev/ai-assistants-getting-started-guide.md (hunk 2 above)
Comment (andriitserkovnyi) [reply docs/dg/dev/ai-assistants-getting-started-guide.md:36]: Does this screenshot show Codiumate? Does it always show up in this menu, regardless of what is highlighted?

Comment (kraal-spryker) [reply docs/dg/dev/ai-assistants-getting-started-guide.md:36]: If codiumate menu exists - it means that plugin is installed successsfully

**Pull request merged**